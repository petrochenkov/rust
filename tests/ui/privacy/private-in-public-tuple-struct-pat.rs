mod m {
    struct Priv;

    #[allow(private_interfaces)]
    pub struct PubStruct(pub Priv);
    pub enum PubEnum {
        #[allow(private_interfaces)]
        Variant(Priv),
    }

    impl PubStruct {
        pub fn new() -> PubStruct {
            PubStruct(Priv)
        }
    }
    impl PubEnum {
        pub fn new() -> PubEnum {
            PubEnum::Variant(Priv)
        }
    }
}

fn main() {
    match m::PubStruct::new() {
        m::PubStruct(_) => {} //~ ERROR type `Priv` is private
        m::PubStruct(..) => {} //~ ERROR type `Priv` is private
    }

    match m::PubEnum::new() {
        m::PubEnum::Variant(_) => {} //~ ERROR type `Priv` is private
        m::PubEnum::Variant(..) => {} //~ ERROR type `Priv` is private
    }

    let _ = m::PubStruct; //~ ERROR type `Priv` is private
    let _ = m::PubEnum::Variant; //~ ERROR type `Priv` is private
}
