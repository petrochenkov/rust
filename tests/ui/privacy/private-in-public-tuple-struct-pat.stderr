error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:25:9
   |
LL |         m::PubStruct(_) => {}
   |         ^^^^^^^^^^^^ private type

error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:26:9
   |
LL |         m::PubStruct(..) => {}
   |         ^^^^^^^^^^^^ private type

error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:30:9
   |
LL |         m::PubEnum::Variant(_) => {}
   |         ^^^^^^^^^^^^^^^^^^^ private type

error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:31:9
   |
LL |         m::PubEnum::Variant(..) => {}
   |         ^^^^^^^^^^^^^^^^^^^ private type

error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:34:13
   |
LL |     let _ = m::PubStruct;
   |             ^^^^^^^^^^^^ private type

error: type `Priv` is private
  --> $DIR/private-in-public-tuple-struct-pat.rs:35:13
   |
LL |     let _ = m::PubEnum::Variant;
   |             ^^^^^^^^^^^^^^^^^^^ private type

error: aborting due to 6 previous errors

